### ISSUE TITLE
System.Net.Security.Tests.NegotiateAuthenticationKerberosTest asserts on Android

### ISSUE THREAD

**Issue by karelz** [dotnet/runtime#74708]
Tests affected:
- **Loopback_Success** - `System.Net.Security.Tests.NegotiateAuthenticationKerberosTest.Loopback_Success`
- **Invalid_Token** - `System.Net.Security.Tests.NegotiateAuthenticationKerberosTest.Invalid_Token`

Failure frequency - results with "Test Results" link in [last 30 days](https://runfo.azurewebsites.net/search/tests/?q=started%3A%7E30+definition%3Aruntime-extra-platforms+name%3A%22system.net.security.tests+work+item%22) - as of 8/27:

Day | Run | Details
-- | -- | --
8/27 | **Rolling run** [1971504](https://runfo.azurewebsites.net/view/build/?number=1971504) | net7.0-Linux-Release-arm64-Mono_Release_LinuxBionic-Windows.10.Amd64.Android.Open
8/26 | **Rolling run** [1970381](https://runfo.azurewebsites.net/view/build/?number=1970381) | net7.0-Linux-Release-arm64-Mono_Release_LinuxBionic-Windows.10.Amd64.Android.Open

- **Loopback_Success** - `System.Net.Security.Tests.NegotiateAuthenticationKerberosTest.Loopback_Success`
```
Microsoft.DotNet.RemoteExecutor.RemoteExecutionException : Remote process failed with an unhandled exception.

Child exception:
  Xunit.Sdk.TrueException: Client authentication failed with Unsupported
Expected: True
Actual:   False
   at System.Net.Security.Tests.NegotiateAuthenticationKerberosTest.<>c.<Loopback_Success>b__2_0() in /_/src/libraries/System.Net.Security/tests/FunctionalTests/NegotiateAuthenticationKerberosTest.cs:line 48
   at System.Reflection.MethodInvoker.InterpretedInvoke(Object obj, Span`1 args, BindingFlags invokeAttr) in /_/src/mono/System.Private.CoreLib/src/System/Reflection/MethodInvoker.Mono.cs:line 33

Child process:
  System.Net.Security.Tests, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51 System.Net.Security.Tests.NegotiateAuthenticationKerberosTest+<>c Void <Loopback_Success>b__2_0()
```
- **Invalid_Token** - `System.Net.Security.Tests.NegotiateAuthenticationKerberosTest.Invalid_Token`
```
Microsoft.DotNet.RemoteExecutor.RemoteExecutionException : Remote process failed with an unhandled exception.

Child exception:
  Xunit.Sdk.EqualException: Assert.Equal() Failure
Expected: ContinueNeeded
Actual:   GenericFailure
   at System.Net.Security.Tests.NegotiateAuthenticationKerberosTest.<>c.<Invalid_Token>b__3_0() in /_/src/libraries/System.Net.Security/tests/FunctionalTests/NegotiateAuthenticationKerberosTest.cs:line 75
   at System.Reflection.MethodInvoker.InterpretedInvoke(Object obj, Span`1 args, BindingFlags invokeAttr) in /_/src/mono/System.Private.CoreLib/src/System/Reflection/MethodInvoker.Mono.cs:line 33

Child process:
  System.Net.Security.Tests, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51 System.Net.Security.Tests.NegotiateAuthenticationKerberosTest+<>c Void <Invalid_Token>b__3_0()
```

cc @filipnavara 

Likely related to #73343 and #74224

#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|

**Comment by filipnavara** [comment]
This is same as #74224. As stated in https://github.com/dotnet/runtime/pull/74225#issuecomment-1220448471 the code is guarded not to execute on Android. We need to figure out what the h*ck is this pipeline and why that check doesn't correctly trigger.

**Comment by filipnavara** [comment]
@akoeplinger Got any idea what run is that and why would `OperatingSystem.IsLinux()` return true for Android? (or does it get incorrectly substituted by linker?)

**Comment by filipnavara** [comment]
Hmm, apparently the Linux Bionic pipeline is something weird that I didn't know...and it's not Android. I can add extra check much like https://github.com/dotnet/runtime/pull/72256/files did.

**Comment by karelz** [comment]
Thanks for figuring it out @filipnavara!

**Comment by karelz** [comment]
Reopening for backport to 7.0.

**Comment by wfurt** [comment]
7.0 is fixed by #75047
